modules/search/class.jetpack-search.php

```diff
@@ -737,38 +784,43 @@ function convert_wp_es_to_es_args( array $args ) {
 		// Filters rock because they are cached from one query to the next
 		// but they are cached as individual filters, rather than all combined together.
 		// May get performance boost by also caching the top level boolean filter too.
-		$filters = array();
 
 		if ( $args['post_type'] ) {
 			if ( ! is_array( $args['post_type'] ) ) {
 				$args['post_type'] = array( $args['post_type'] );
 			}
 
-			$filters[] = array(
+			$parser->add_filter( array(
 				'terms' => array(
 					'post_type' => $args['post_type'],
 				),
-			);
+			) );
 		}
 
 		if ( $args['author_name'] ) {
-			$filters[] = array(
-				'terms' => array(
-					'author_login' => $args['author_name'],
-				),
-			);
+			/*
+
+			@TODO!
+
+			$parser->author_field_filter( array(
```

Comment:
let's leave off the author filtering based on the search string for this PR.

Phrase filtering is the bigger feature. I'm really not sure how much end users organically use @ and # in their quires to mean something. The author field filter is so much built for WP.com that I am not sure it will work the way we hope.

So this just becomes a clause to add a filter on author_login when one gets specified.